R/checkers.R

```diff
@@ -317,3 +326,173 @@ assert_single_value <- function(x, arg = "x",
   }
   invisible()
 }
+
+#' Assert that the dataframe being passed to the function is not empty
+#'
+#' @param x the dataframe to check
+#' @param arg the name of the dataframe to check
+#' @param call Calling environment to be passed to [cli::cli_abort()] for
+#' traceback.
+#' @param add_err_msg add_err_msg string containing an additional error message,
+#' default is the empty string (`""`)
+#'
+#' @return NULL invisible
+assert_not_empty <- function(x, arg = "x",
+                             call = rlang::caller_env(),
+                             add_err_msg = "") {
+  if (nrow(x) == 0) {
+    cli::cli_abort(
+      c(
+        "{.arg {arg}} is empty",
+        add_err_msg
+      ),
+      call = call,
+      class = "wwinference_input_data_error"
+    )
+  }
+  invisible()
+}
+
+
+#' Assert that the vector of dates being passed in contains dates for each day
+#'
+#' @param dates the vector of dates to check, must be of Date type
+#' @param call Calling environment to be passed to [cli::cli_abort()] for
+#' traceback.
+#' @param add_err_msg add_err_msg string containing an additional error message,
+#' default is the empty string (`""`)
+#'
+#' @return NULL invisible
+assert_daily_data <- function(dates,
+                              call = rlang::caller_env(),
+                              add_err_msg = "") {
+  sorted_dates <- dates[order(dates)]
+  # Generate a sequence of dates from the minimum to the
+  # maximum date in the dataset
+  expected_dates <- seq.Date(
+    from = min(sorted_dates),
+    to = max(sorted_dates),
+    by = "day"
+  )
+
+
+  if (!all(expected_dates %in% sorted_dates)) {
+    cli::cli_abort(
+      c(
+        "Vector of dates does not contain dates for each day",
+        add_err_msg
+      ),
+      call = call,
+      class = "wwinference_input_data_error"
+    )
+  }
+  invisible()
+}
+
+#' Assert that the vector of dates contains the duration of the specified
+#' calibration time
+#'
+#' @param date_vector the vector of dates to check, must be of Date type
+#' @param calibration_time integer indicating the number of days that
+#' the dates must span
+#' @param call Calling environment to be passed to [cli::cli_abort()] for
+#' traceback.
+#' @param add_err_msg add_err_msg string containing an additional error message,
+#' default is the empty string (`""`)
+#'
+#' @return NULL invisible
+assert_sufficient_days_of_data <- function(date_vector,
```

Comment:
Do you want to enforce dailiness here too? i.e. once it has verified that the data has a date before or equal to the calibration start, should it call `assert_daily_data` on the calibration period?

Reply:
I don't think so no, I want to keep these checks as separate as possible because for example, if we add a feature for weekly data, we can still have the user specify a calibration time in days, and then this will just check that theres enough data for that calibration time. A separate check will check for the daily nature of the data, and we can remove it for that use case. 

I'd rather keep these checks/asserts as modular as possible so that when we do make changes to the input data we know which ones we expect to fail.